File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.cpp
// Last updated: 6/11/2026, 11:36:12 AM
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, int> ump;
        for(int el: nums) ump[el] = 1;
        int ans = 0;
        for(auto elem: ump){
            int el = elem.first;
            if(ump.find(el-1) != ump.end()) continue;
            int cur = el;
            int cnt = 0;
            while(1){
                if(ump.find(cur) != ump.end()){
                    cur++;
                    cnt++;
                }else{
                    break;
                }
            }
            ans = max(ans, cnt);
        }
        return ans;
    }
};

File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.py
# Last updated: 6/11/2026, 11:36:16 AM
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        set_nums = set(nums)
       
        maxct = 0
        for i in range(len(nums)):
            ct = 0
            if(nums[i]-1 not in set_nums):
                cur = nums[i]
                ct = 1
                while(cur+1 in set_nums):
                    cur = cur+1
                    ct+=1
                maxct = max(maxct, ct)
        return maxct
